### ISSUE TITLE
Potentially faulty core fixes: array indentation / array interspersed with comments

### ISSUE THREAD

**Issue by jrfnl** [WordPress-Coding-Standards/WordPress-Coding-Standards#985]
#### Summary:
While looking at the array indentation sniff, I realized that if there are comments between array items, the indentation of these would not be corrected.

Related #973

#### Example:

File: N/A - I have no doubt, we will actually run into this, but haven't come across it yet in the limited group of fixed files I've inspected so far.

Source:
```php
get_current_screen()->add_help_tab( array(
'id'		=> 'overview',
// Here we have a comment.
'title'		=> __('Overview'),
// Here we have a comment.
'content'	=> 'value',
) );
```

Fixed as:
```php
get_current_screen()->add_help_tab( array(
	'id'		=> 'overview',
// Here we have a comment.
	'title'		=> __('Overview'),
// Here we have a comment.
	'content'	=> 'value',
) );
```

Expected fix:
```php
get_current_screen()->add_help_tab( array(
	'id'		=> 'overview',
	// Here we have a comment.
	'title'		=> __('Overview'),
	// Here we have a comment.
	'content'	=> 'value',
) );
```

**Comment by GaryJones** [comment]
Not sure if it's applicable here, but tests may also want to cover a DocBlock as well as single line comments i.e.

```php
$my_array = [
	/**
	 * Documentation about a filtered value...
	 */
	'truthy' = apply_filters( '...', true );
]
```

**Comment by ntwb** [comment]
Via https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#5-2-multi-line-comments
> _"Important note: Multi-line comments must not begin with `/**` (double asterisk) as the parser might mistake it for a DocBlock. Use `/*` (single asterisk) instead."_

So, updating the example above so that is *not* detected as a DocBlock:
```php
$my_array = [
	/*
	 * Documentation about a filtered value...
	 */
	'truthy' = apply_filters( '...', true );
]
```

**Comment by jrfnl** [comment]
@ntwb Sniffs should function independently of each other. So both those comment styles should be detected.

P.S.: both the above examples have a parse error: `=` should be `=>` and the `;` should be `,`.

FYI: I've been working on a fix for this and #973 and believe I have got this sorted, but running into some issues with differences in the end results between the unit tests and manual runs.

**Comment by ntwb** [comment]
Right, thanks @jrfnl, still much to learn in the art of sniffing 😏
